src/derive/field.rs

```diff
@@ -365,6 +412,27 @@ macro_rules! field_arithmetic {
 
                 $field([d0 & mask, d1 & mask, d2 & mask, d3 & mask])
             }
+
+            /// Lexicographic comparison of Montgomery forms.
+            #[inline(always)]
+            fn is_less_than(x: &[u64; 4], y: &[u64; 4]) -> bool {
```

Comment:
Great suggestion! I actually used `is_less_than` because I had some anecdotal impression the branching would make it on average faster. But I just added a criterion benchmark and the results speak for themselves:
```
Big less than methods/is_less_than/                        
                        time:   [572.38 ps 573.28 ps 574.40 ps]
                        change: [-0.4200% -0.0114% +0.3645%] (p = 0.96 > 0.05)
                        No change in performance detected.
Found 10 outliers among 100 measurements (10.00%)
  1 (1.00%) high mild
  9 (9.00%) high severe
Benchmarking Big less than methods/check_underflow/: Warming up for 3.00                                                                        Benchmarking Big less than methods/check_underflow/: Collecting 100 samp                                                                        Big less than methods/check_underflow/                        
                        time:   [286.14 ps 286.48 ps 286.90 ps]
                        change: [-0.3090% +0.2068% +0.7208%] (p = 0.46 > 0.05)
                        No change in performance detected.
Found 11 outliers among 100 measurements (11.00%)
  1 (1.00%) high mild
  10 (10.00%) high severe
```

This is very helpful to know.